### PR TITLE
feat!(NODE-4862): add BSONType enum and remove internal constants from export

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- removes internal constants from export, 
- Adds the BSONType enum from the driver

#### What is the motivation for this change?

The exports are a blocker for #524 passing typescript tests

The enum is a nice to have, it provides the same information that constants did but in a consistent way the driver already does, locating the enum here puts it closer to the code that it relates to, its not used by the BSON library but if someone needs type indicators this is a consistent way to make it available. 

### Double check the following

- [x] Ran `npm run lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/node/constants.test.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    });
+    it('Min key should be 0xFF', () => {
+      expect(constants.BSON_DATA_MIN_KEY).to.equal(0xff);
+    });
+    it('Max key should be 0x7F', () => {
+      expect(constants.BSON_DATA_MAX_KEY).to.equal(0x7f);
+    });
+  });
+
+  describe('BSONType enum', () => {
```
Comment (nbbeeken) [inline test/node/constants.test.ts:143]: Tests above are old, no change except for the import
Tests here are new for the enum

File: src/constants.ts [hunk 2]
```diff
@@ -108,3 +108,31 @@ export const BSON_BINARY_SUBTYPE_COLUMN = 7;
...
+  int: 16,
+  timestamp: 17,
+  long: 18,
+  decimal: 19,
+  minKey: -1,
+  maxKey: 127
+} as const);
+
+/** @public */
+export type BSONType = typeof BSONType[keyof typeof BSONType];
```
Comment (nbbeeken) [inline src/constants.ts:138]: Using the enum pattern here, it makes BSONType the annotation equal to a union of the values.
BSONType the runtime value is an object with properties.

**Review by durran — COMMENTED** [review]
Are we planning on removing the export in https://github.com/mongodb/node-mongodb-native/blob/main/src/mongo_types.ts to just re-export `BSONType` from here as part of NODE-4701?

**Review by durran — APPROVED** [review]

**Comment by nbbeeken** [comment]
I expect we'll change to a re-export in this ticket: https://jira.mongodb.org/browse/NODE-4867

**Review by durran — APPROVED** [review]

**Pull request merged**